src/components/HomeLandingPage/HomeLandingPage.tsx

```diff
@@ -40,24 +43,76 @@ const HomeLandingPage = () => {
   useAllEvents();
 
   // Fetch events from cache
-  const { data } = useQuery(GET_ALL_EVENTS);
+  const { data } = useQuery(FETCH_ALL_EVENTS);
+  const [editEvent] = useMutation(EDIT_EVENT);
+  const [deleteEvent] = useMutation(DELETE_EVENT, {
+    update(cache, { data: { deleteEvent } }) {
+      if (!deleteEvent) {
+        return;
+      }
+      let { events } = cache.readQuery<any>({
+        query: FETCH_ALL_EVENTS,
+      });
+
+      events = events.filter((event) => event.id !== selectedEvent?.id);
+
+      cache.writeQuery({
+        query: FETCH_ALL_EVENTS,
+        data: { events },
+      });
+    },
+  });
   const events: Array<Event> = data ? data.events : [];
 
+  useEffect(() => {
+    if (selectedEvent) {
+      deleteEvent({
+        variables: {
+          id: selectedEvent?.id,
+        },
+      });
+    }
```

Comment:
Curious, what is your reasoning for doing this in `selectedEvent`? This doesn't really seem like the place for a `delete`. Also it looks like `selectedEvent` is only being used for delete. Can you name it more clearly if that is the case? Something like `eventToBeDeleted`?

Reply:
Yeah good point about the naming - I'll rename it to `eventToBeDeleted` for clarity. The reason why I need this is because I needed `selectedEvent` to be set before I called the `deleteEvent` mutation - so I added it to this `useEffect` call. The async/await I had before was for this purpose, and I refactored it what it is now.

Reply:
`selectedEvent` is used in the `deleteEvent` callback to properly update the cache